### PR TITLE
Fix/non centos x509tests

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- fixed x509 set up for windows and mac
- fixed ssl for windows and mac

### CODE REVIEW

File: libraries/provision/ansible/playbooks/tasks/deploy-sync-gateway-config-macos.yml [hunk 1]
```diff
@@ -35,6 +35,11 @@
   when: x509_auth
   become: yes
 
+- name: SYNC GATEWAY | Change the mode permissions
+  become: yes
+  shell: chmod 777 /Users/sync_gateway/certs/*.*
```
Comment (mgcouchbase) [inline libraries/provision/ansible/playbooks/tasks/deploy-sync-gateway-config-macos.yml:40]: I like this

File: resources/x509_cert_gen/gen_keystore.sh [hunk 2]
```diff
@@ -65,20 +65,21 @@ do
...
 	# Upload ROOT CA and activate it
 	curl -s -o /dev/null --data-binary "@./${ROOT_CA}.pem" \
     	http://${ADMINCRED}@${ip}:8091/controller/uploadClusterCA
 	curl -sX POST http://${ADMINCRED}@${ip}:8091/node/controller/reloadCertificate
+
 	# Enable client cert
 	if ${USE_JSON} ;then
-        	POST_DATA='{"state": "enable","prefixes": [{"path": "subject.cn","prefix": "","delimiter": ""}]}'
-        	curl -s -H "Content-Type: application/json" -X POST -d "${POST_DATA}" http://${ADMINCRED}@${ip}:8091/settings/clientCertAuth
+	    POST_DATA='{"state": "enable","prefixes": [{"path": "subject.cn","prefix": "","delimiter": ""}]}'
```
Comment (mgcouchbase) [inline resources/x509_cert_gen/gen_keystore.sh:78]: Can you fix the alignment here

Comment (sridevi-15) [reply resources/x509_cert_gen/gen_keystore.sh:78]: actullay alignment is fixed already. Earlier, it use to have two tabs after if, now it has one tab. That is the one you are mentioning?

File: libraries/provision/ansible/playbooks/tasks/deploy-sync-gateway-config-macos.yml (hunk 1 above)
Comment (sridevi-15) [reply libraries/provision/ansible/playbooks/tasks/deploy-sync-gateway-config-macos.yml:40]: this is the one u might need it

File: resources/x509_cert_gen/gen_keystore.sh (hunk 2 above)
Comment (mgcouchbase) [inline resources/x509_cert_gen/gen_keystore.sh:78]: I see 78 and 79 line are not starting at same position

Comment (sridevi-15) [reply resources/x509_cert_gen/gen_keystore.sh:78]: done

**Pull request merged**